napari/layers/utils/_tests/test_text_manager.py

```diff
@@ -544,3 +551,161 @@ def test_view_text_with_constant_text():
     # because assert_array_equal broadcasts scalars automatically
     assert len(actual) == 2
     np.testing.assert_array_equal(actual, ['A', 'A'])
+
+
+def test_init_with_constant_color():
+    color = {'constant': 'red'}
+    features = pd.DataFrame(index=range(3))
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, 'red')
+
+
+def test_init_with_manual_color():
+    color = ['red', 'green', 'blue']
+    features = pd.DataFrame({'class': ['A', 'B', 'C']})
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['red', 'green', 'blue'])
+
+
+def test_init_with_derived_color():
+    color = {'feature': 'colors'}
+    features = pd.DataFrame({'colors': ['red', 'green', 'blue']})
+
+    text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['red', 'green', 'blue'])
+
+
+def test_init_with_derived_color_missing_feature():
+    color = {'feature': 'not_a_feature'}
+    features = pd.DataFrame({'colors': ['red', 'green', 'blue']})
+
+    with pytest.warns(RuntimeWarning):
+        text_manager = TextManager(color=color, features=features)
+
+    actual = text_manager.color._values
+    assert_colors_equal(actual, ['cyan'] * 3)
```

Comment:
is cyan the default colour or something? if so pls explain in a comment

Reply:
Good catch! It's actually the fallback color that's used when the encoding fails. Instead of adding a comment and/or using the constant (which is the default fallback), I updated the test name to indicate the expected behavior here and specified the fallback in the input so that the expected values are more clear.